src/VisualStudio/Core/Def/Implementation/TaskList/CommentTaskTokenSerializerProvider.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using IAsyncServiceProvider = Microsoft.VisualStudio.Shell.IAsyncServiceProvider;
+using ITaskList = Microsoft.VisualStudio.Shell.ITaskList;
+using SAsyncServiceProvider = Microsoft.VisualStudio.Shell.Interop.SAsyncServiceProvider;
+using SVsTaskList = Microsoft.VisualStudio.Shell.Interop.SVsTaskList;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.TaskList
+{
+    [Export(typeof(IOptionPersisterProvider))]
+    internal sealed class CommentTaskTokenSerializerProvider : IOptionPersisterProvider
```

Comment:
All of these `IOptionPersisterProvider`'s look very similar, can we combine most of this into a common base class?

Reply:
This was covered by https://github.com/dotnet/roslyn/pull/52063#discussion_r600014550. In this case, sharing doesn't bring enough value to cover its own complexity cost.